### ISSUE TITLE
Add github actions to run tests

### ISSUE THREAD

**Issue by mdekrey** [PrincipleStudios/scalable-git-branching-tools#12]
Add GitHub actions to run tests on pull requests. It should:

1. Run `Invoke-Pester` and verify all successes.
2. Run the `docker build .` to verify the tests that were dockerized for E2E git testing.

These could be two separate actions to run in parallel.

Pester can also produce code coverage; consider having it output the result somewhere we can review it, such as attached to the PR. If the solution does not include this, fork this request to a new ticket.
